### PR TITLE
[ADVAPP-665]: enhance failure engagements notifications

### PR DESCRIPTION
**Pull request by jayushi-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-665

### Technical Description

> Add notification type in notification when email is failed.

### CODE REVIEW

File: app-modules/engagement/src/Notifications/EngagementFailedNotification.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ public function toDatabase(object $notifiable): array
         return FilamentNotification::make()
             ->danger()
             ->title('Engagement Delivery Failed')
-            ->body("Your engagement failed to be delivered to {$this->engagement->recipient->display_name}.")
+            ->body("Your engagement email failed to be delivered to {$this->engagement->recipient->display_name}.")
```
Comment (Orrison) [inline app-modules/engagement/src/Notifications/EngagementFailedNotification.php:75]: As discussed, let's make sure to say `sms` or `email` here depending on the type of the notification that failed.

**Review by Orrison — CHANGES_REQUESTED** [review]

**Comment by Orrison** [comment]
@jayushi-canyon just some feedback we already discussed. Also take not of the changes I made to the PR title and description as that is the proper format to make sure our PRs are created in.

File: app-modules/engagement/src/Notifications/EngagementFailedNotification.php (hunk 1 above)
Comment (jayushi-canyon) [reply app-modules/engagement/src/Notifications/EngagementFailedNotification.php]: @Orrison Made the changes.

File: app-modules/engagement/src/Notifications/EngagementFailedNotification.php [hunk 2]
```diff
@@ -69,10 +70,18 @@ public function toEmail(object $notifiable): MailMessage
...
+            $engagementType = $this->engagement->deliverable->channe;
+        } elseif ($this->engagement->deliverable->channel == EngagementDeliveryMethod::Sms) {
+            $engagementType = 'SMS';
+        }
+
         return FilamentNotification::make()
             ->danger()
             ->title('Engagement Delivery Failed')
-            ->body("Your engagement failed to be delivered to {$this->engagement->recipient->display_name}.")
+            ->body("Your engagement { {$engagementType}} failed to be delivered to {$this->engagement->recipient->display_name}.")
```
Comment (Orrison) [inline app-modules/engagement/src/Notifications/EngagementFailedNotification.php:84]:
```suggestion
            ->body("Your engagement {$engagementType} failed to be delivered to {$this->engagement->recipient->display_name}.")
```

File: app-modules/engagement/src/Notifications/EngagementFailedNotification.php [hunk 3]
```diff
@@ -69,10 +70,18 @@ public function toEmail(object $notifiable): MailMessage
 
     public function toDatabase(object $notifiable): array
     {
+        $engagementType = '';
+
+        if ($this->engagement->deliverable->channel == EngagementDeliveryMethod::Email) {
+            $engagementType = $this->engagement->deliverable->channe;
+        } elseif ($this->engagement->deliverable->channel == EngagementDeliveryMethod::Sms) {
+            $engagementType = 'SMS';
+        }
```
Comment (Orrison) [inline app-modules/engagement/src/Notifications/EngagementFailedNotification.php:79]: A `match` statement would probably work better here.

Also the results for the email type seems incomplete.

**Review by Orrison — CHANGES_REQUESTED** [review]

**Comment by jayushi-canyon** [comment]
> @jayushi-canyon just some feedback we already discussed. Also take not of the changes I made to the PR title and description as that is the proper format to make sure our PRs are created in.

@Orrison Noted!

File: app-modules/engagement/src/Notifications/EngagementFailedNotification.php (hunk 2 above)
Comment (jayushi-canyon) [reply app-modules/engagement/src/Notifications/EngagementFailedNotification.php]: @Orrison Done

File: app-modules/engagement/src/Notifications/EngagementFailedNotification.php (hunk 3 above)
Comment (jayushi-canyon) [reply app-modules/engagement/src/Notifications/EngagementFailedNotification.php]: @Orrison Made the changes

**Pull request closed without merging**